### PR TITLE
Batchedblas wording sketch

### PR DESCRIPTION
**Pull request by crtrott**

### CODE REVIEW

File: batched_blas/batched_blas.md [hunk 1]
```diff
@@ -325,6 +325,110 @@ have a natural representation without them.
...
+
+template<@_batched-out-vector_@ OutVec>
+auto get_batch(OutVec v, int batch) { return submdspan(v, batch, full_extent); }
+```
+
+With those functions we can now define the batched overload for `matrix_vector_product`:
+
+```c++
+template<class ExecutionPolicy,
+         @_batched-in-matrix_@ InMat,
```
Comment (dalg24) [inline batched_blas/batched_blas.md:398]: is not defined

File: batched_blas/batched_blas.md [hunk 2]
```diff
@@ -325,6 +325,110 @@ have a natural representation without them.
...
+         @_batched-in-vector_@ InVec,
+         @_batched-out-vector_@ OutVec>
+void matrix_vector_product(ExecutionPolicy&& exec,
+                           InMat A,
+                           InVec x,
+                           OutVec y);
+```
+
+*Preconditions:* `A.extent(0) == x.extent(0) && A.extent(0) == y.extent(0)` is `true`.
```
Comment (dalg24) [inline batched_blas/batched_blas.md:407]: does it not depend on the ranks of A and x?

File: batched_blas/batched_blas.md [hunk 3]
```diff
@@ -325,6 +325,110 @@ have a natural representation without them.
...
+template<@_batched-in-vector_@ InVec>
+requires(InVec::rank()==1)
+auto get_batch(InVec v, int /*batch*/) { return v; }
+
+template<@_batched-in-vector_@ InVec>
+requires(InVec::rank()==2)
+auto get_batch(InVec v, int batch) { return submdspan(v, batch, full_extent); }
+
+template<@_batched-out-vector_@ OutVec>
+auto get_batch(OutVec v, int batch) { return submdspan(v, batch, full_extent); }
```
Comment (dalg24) [inline batched_blas/batched_blas.md:391]: Assuming that you define `@_batched-in-matrix_@` somewhere, I don't see how you can stir A passed as rank-2 and x passed at rank-2 towards different overloads of `get_batch`.

File: batched_blas/batched_blas.md [hunk 4]
```diff
@@ -325,6 +325,110 @@ have a natural representation without them.
...
+void matrix_vector_product(ExecutionPolicy&& exec,
+                           Alphas alphas,
+                           InMat A,
+                           InVec x,
+                           OutVec y);
+```
+
+*Preconditions:* `A.extent(0) == x.extent(0) && A.extent(0) == y.extent(0) && A.extent(0) == alphas.extent(0)` is `true`.
+
+*Effects:* Equivalent to calling `matrix_vector_product(scaled(alphas[i],get_batch(A, i)), get_batch(x,  i), get_batch(y, i))` for each `i` in the range of $[$ 0,`A.extent(0)`$)$.
```
Comment (dalg24) [inline batched_blas/batched_blas.md:429]: I don't think this works.

File: batched_blas/batched_blas.md (hunk 2 above)
Comment (crtrott) [reply batched_blas/batched_blas.md:407]: no

File: batched_blas/batched_blas.md (hunk 1 above)
Comment (crtrott) [reply batched_blas/batched_blas.md:398]: this is a sketch where I did not define everything, but imply analogous definitions

File: batched_blas/batched_blas.md (hunk 4 above)
Comment (crtrott) [reply batched_blas/batched_blas.md:429]: as in you don't think this is the right way to word it?

File: batched_blas/batched_blas.md (hunk 3 above)
Comment (crtrott) [reply batched_blas/batched_blas.md:391]: yeah you are right, needs a name `get_batch_vector`, `get_batch_matrix` ...

File: batched_blas/batched_blas.md (hunk 2 above)
Comment (crtrott) [reply batched_blas/batched_blas.md:407]: `(y.extent(0) == x.extent(0) || x.rank()==1) && (y.extent(0) == A.extent(0) || A.rank()==2)`

File: batched_blas/batched_blas.md (hunk 4 above)
Comment (dalg24) [reply batched_blas/batched_blas.md]: It was related to my comments above about overload resolution and get_batch, but I see your updated version has get_batch_{matrix,vector} so it looks better.

**Pull request merged**